include/bluetooth/services/ots_srv.h

```diff
@@ -0,0 +1,641 @@
+	((feat) & BIT(BT_GATT_OTS_SRV_OACP_FEAT_PATCH))
+
+/** @brief Get @ref BT_GATT_OTS_SRV_OACP_FEAT_ABORT feature.
+ *
+ *  @param feat OTS features.
+ */
+#define BT_GATT_OTS_SRV_OACP_GET_FEAT_ABORT(feat) \
+	((feat) & BIT(BT_GATT_OTS_SRV_OACP_FEAT_ABORT))
+
+/** @brief Object List Control Point Feature bits. */
+enum {
+	/** Bit 0 OLCP Go To Op Code Supported */
+	BT_GATT_OTS_SRV_OLCP_FEAT_GO_TO      = 0,
+
+	/** Bit 1 OLCP Order Op Code Supported */
+	BT_GATT_OTS_SRV_OLCP_FEAT_ORDER      = 1,
+
+	/** Bit 2 OLCP Request Number of Objects Op Code Supported */
+	BT_GATT_OTS_SRV_OLCP_FEAT_NUM_REQ    = 2,
+
+	/** Bit 3 OLCP Clear Marking Op Code Supported*/
+	BT_GATT_OTS_SRV_OLCP_FEAT_CLEAR      = 3,
+};
+
+/** @brief Set @ref BT_GATT_OTS_SRV_OLCP_FEAT_GO_TO feature.
+ *
+ *  @param feat OTS features.
+ */
+#define BT_GATT_OTS_SRV_OLCP_SET_FEAT_GO_TO(feat) \
+	WRITE_BIT(feat, BT_GATT_OTS_SRV_OLCP_FEAT_GO_TO, 1)
+
+/** @brief Set @ref BT_GATT_OTS_SRV_OLCP_FEAT_ORDER feature.
+ *
+ *  @param feat OTS features.
+ */
+#define BT_GATT_OTS_SRV_OLCP_SET_FEAT_ORDER(feat) \
+	WRITE_BIT(feat, BT_GATT_OTS_SRV_OLCP_FEAT_ORDER, 1)
+
+/** @brief Set @ref BT_GATT_OTS_SRV_OLCP_FEAT_NUM_REQ feature.
+ *
+ *  @param feat OTS features.
+ */
+#define BT_GATT_OTS_SRV_OLCP_SET_FEAT_NUM_REQ(feat) \
+	WRITE_BIT(feat, BT_GATT_OTS_SRV_OLCP_FEAT_NUM_REQ, 1)
+
+/** @brief Set @ref BT_GATT_OTS_SRV_OLCP_FEAT_CLEAR feature.
+ *
+ *  @param feat OTS features.
+ */
+#define BT_GATT_OTS_SRV_OLCP_SET_FEAT_CLEAR(feat) \
+	WRITE_BIT(feat, BT_GATT_OTS_SRV_OLCP_FEAT_CLEAR, 1)
+
+/** @brief Get @ref BT_GATT_OTS_SRV_OLCP_GET_FEAT_GO_TO feature.
+ *
+ *  @param feat OTS features.
+ */
+#define BT_GATT_OTS_SRV_OLCP_GET_FEAT_GO_TO(feat) \
+	((feat) & BIT(BT_GATT_OTS_SRV_OLCP_FEAT_GO_TO))
+
+/** @brief Get @ref BT_GATT_OTS_SRV_OLCP_GET_FEAT_ORDER feature.
+ *
+ *  @param feat OTS features.
+ */
+#define BT_GATT_OTS_SRV_OLCP_GET_FEAT_ORDER(feat) \
+	((feat) & BIT(BT_GATT_OTS_SRV_OLCP_FEAT_ORDER))
+
+/** @brief Get @ref BT_GATT_OTS_SRV_OLCP_GET_FEAT_NUM_REQ feature.
+ *
+ *  @param feat OTS features.
+ */
+#define BT_GATT_OTS_SRV_OLCP_GET_FEAT_NUM_REQ(feat) \
+	((feat) & BIT(BT_GATT_OTS_SRV_OLCP_FEAT_NUM_REQ))
+
+/** @brief Get @ref BT_GATT_OTS_SRV_OLCP_GET_FEAT_CLEAR feature.
+ *
+ *  @param feat OTS features.
+ */
+#define BT_GATT_OTS_SRV_OLCP_GET_FEAT_CLEAR(feat) \
+	((feat) & BIT(BT_GATT_OTS_SRV_OLCP_FEAT_CLEAR))
+
+/**@brief Features of the OTS. */
+struct bt_gatt_ots_srv_feat {
+	/* OACP Features */
+	uint32_t oacp;
+
+	/* OLCP Features */
+	uint32_t olcp;
+};
+
+/** @brief Opaque OTS instance. */
+struct bt_gatt_ots_srv;
+
+/** @brief OTS server callback structure. */
+struct bt_gatt_ots_srv_cb {
+	/** Object created callback
+	 *
+	 *  This callback is called whenever a new object is created.
+	 *  Application can reject this request by returning an error
+	 *  when it does not have necessary resources to hold this new
+	 *  object.
+	 *
+	 *  @param ots  OTS Server instance.
```

Comment:
I am not sure about it, but you are right that the server is single entity.

Let's wait for feedback from @joerchan, because this will set a precedent on how we document remaining services that are going to be aligned with new naming convention (_srv).

How about this?
```suggestion
	 *  @param ots  Server-side OTS instance.
```

Reply:
I would actually suggest to just have 
```
@param ots  OTS instance.
```

As that is, to me at least, very descriptive of what it is - A OT service instance

Reply:
I'm not following. there can be many instances of `bt_gatt_ots_srv`, right? You can get one from the pool using `bt_gatt_ots_srv_free_instance_get`
I think it makes sense to include that it is an OTS server instance, since we could also have OTS client instances in the future.

Reply:
Isn't OTP the client side of here? Not sure why we would want to mix the terminologies of service with client, at least with regards to GATT a service already means a server so saying client instance here kind goes against that.

Reply:
I guess it's a terminology discussion by now :)

My understanding is that there is 1 server, which may have one or more _service_ instances. 

Similarly on the client side: There's 1 client, but the client may interact with one or more _service_ instances.

Reply:
> 
> 
> I'm not following. there can be many instances of `bt_gatt_ots_srv`, right? You can get one from the pool using `bt_gatt_ots_srv_free_instance_get`

Yes, that's right.






> 
> 
> Isn't OTP the client side of here? Not sure why we would want to mix the terminologies of service with client, at least with regards to GATT a service already means a server so saying client instance here kind goes against that.

Yes, as according to OTP specification. However, we would like to develop client libraries using per service approach. For instance, HRP Client could be divided into dis_cli and hrs_cli instead of hrp_cli that would cover client-side operations for both services. This way, we can use these service client libraries in other profiles, because they are more modular. Also, the same service could be included in many profiles (e.g DIS in HID over GATT Profile and Heart Rate Profile). Apart from that, many simple services don't have profile specifications (e.g. DIS).

API naming was discussed in this conversation thread:
https://github.com/zephyrproject-rtos/zephyr/pull/27088#discussion_r459370693
and it was changed in this PR as we decided to follow the naming convention used in Mesh (health_cli & health_srv libraries). I hope that's okay for everyone.

Reply:
It's a bit confusing since we have multiple threads discussing naming now :) 

If we want to use the same naming convention as Mesh, where it's named health_cli and health_src, perhaps it should be object (or simply obj) server and client, e.g. `obj_cli` and `obj_srv` (or `object_cli` and `object_srv`), which incidentally also matches the naming convention used in the OTP spec (see https://github.com/zephyrproject-rtos/zephyr/pull/27088#discussion_r481818288).

Reply:
It is "OTS instance" now, resolving